back-end/API/app/services/user_service.py

```
@@ -1,144 +1,90 @@
-            auth_repo.updateActive(user, not user.active)
-        else:
-            logger.warning("Admin (" + current_user.email + ") tried to desactivate itself")
-
+    def update_reset_password(self, email, new_password):
+        user = user_repo.find_by_email(email)
+        user.password = hasher.hash(new_password)
+        user_repo.update(user)
+
+    def update_password(self, current_user, dto: UserUpdatePasswordDTO) -> UserReadDTO:
+        if not current_user.isModerator and current_user.id != dto.id:
+            raise PermissionException(f"L'utilisateur {current_user.id} n'a pas la permission de modifier le mot de passe de l'utilisateur {user.id}")
```

Ruff (0.15.13)

[warning] Avoid specifying long messages outside the exception class

(TRY003)